stacks-signer/src/signerdb.rs

```diff
@@ -197,19 +229,66 @@ static DROP_SCHEMA_0: &str = "
    DROP TABLE IF EXISTS blocks;
    DROP TABLE IF EXISTS db_config;";
 
+static DROP_SCHEMA_1: &str = "
+   DROP TABLE IF EXISTS burn_blocks;
+   DROP TABLE IF EXISTS signer_states;
+   DROP TABLE IF EXISTS blocks;
+   DROP TABLE IF EXISTS db_config;";
+
+static CREATE_BLOCKS_TABLE_2: &str = "
+CREATE TABLE IF NOT EXISTS blocks (
+    reward_cycle INTEGER NOT NULL,
+    signer_signature_hash TEXT NOT NULL,
+    block_info TEXT NOT NULL,
+    consensus_hash TEXT NOT NULL,
+    signed_over INTEGER NOT NULL,
+    broadcasted INTEGER NOT NULL,
+    stacks_height INTEGER NOT NULL,
+    burn_block_height INTEGER NOT NULL,
+    PRIMARY KEY (reward_cycle, signer_signature_hash)
+) STRICT";
+
+static CREATE_BLOCK_SIGNATURES_TABLE: &str = r#"
+CREATE TABLE IF NOT EXISTS block_signatures (
+    -- The block sighash commits to all of the stacks and burnchain state as of its parent,
+    -- as well as the tenure itself so there's no need to include the reward cycle.  Just
+    -- the sighash is sufficient to uniquely identify the block across all burnchain, PoX,
+    -- and stacks forks.
+    signer_signature_hash TEXT NOT NULL,
+    -- signtaure itself
+    signature TEXT NOT NULL,
+    PRIMARY KEY (signature)
+) STRICT;"#;
+
+static CREATE_INDEXES_2: &str = r#"
+CREATE INDEX IF NOT EXISTS block_reward_cycle_and_signature ON block_signatures(signer_signature_hash);
+"#;
+
```

Comment:
I'm not sure it matters, since the system isn't released yet.  Realistically, we don't need migration logic anywhere in the signer.

Reply:
Yeah, I don't think it matters whether there's a DROP or ALTER. There does need to be some logic around this, though, because signers running in mock signing mode may want to upgrade a version without having to manually clear their data directories.

The only point I'd make is that it would be good to have some hygiene here so that we don't end up just having 3 copy-pasted schemas (maybe change the prior schemas to just be empty as long as the DROPs are being executed anyways?)